packages/transactions/src/signer.ts

```diff
@@ -19,6 +19,26 @@ export class TransactionSigner {
     this.originDone = false;
     this.checkOversign = true;
     this.checkOverlap = true;
+
+    // If multi-sig spending condition exists, iterate over
+    // auth fields and reconstruct sigHash
+    let spendingCondition = transaction.auth.spendingCondition;
```

Comment:
I'm curious, can this loop ever fail?

Reply:
yes, it can fail if the auth fields are invalid

Reply:
Can you test the error cases? And document what errors this constructor will throw in the constructor comment?

On that note, does this codebase have any policy of not failing in a constructor?

Reply:
> And document what errors this constructor will throw in the constructor comment

This is not very common in js. I'd consider it a nice-to-have.

> On that note, does this codebase have any policy of not failing in a constructor?

No.